### ISSUE TITLE
[clang-tidy] bugprone-tagged-union-member-count false warning when using enum & pthread_mutex_t

### ISSUE THREAD

**Issue by abdelmaged** [llvm/llvm-project#134840]
Source C file:
```c
#include <pthread.h>

typedef enum {
  MYENUM_ONE,
  MYENUM_TWO,
} myEnumT;

typedef struct {
  pthread_mutex_t mtx;       
  myEnumT         my_enum;
} myTypeT;
```
Using clang-tidy bugprone-tagged-union-member-count:
```
<source>:8:9: warning: tagged union has more data members (3) than tags (2)! [bugprone-tagged-union-member-count]
    8 | typedef struct {
      |         ^
1 warning generated.
```

This is incorrect warning.

clang-tidy vesion: 20.1.1

Issue also reproducible in this link:
https://godbolt.org/z/o3fb9bbrb

**Comment by vbvictor** [comment]
What a coincidence that `pthread_mutex_t` is actually a union with 3 fields
```cpp
typedef union
{
  struct __pthread_mutex_s __data;
  char __size[__SIZEOF_PTHREAD_MUTEX_T];
  long int __align;
} pthread_mutex_t;
```

I think to avoid such problem, we should exclude unions/structs that come from system headers.

**Comment by vbvictor** [comment]
CC @tigbr, you may be interested in this issue

**Comment by tigbr** [comment]
Hi @abdelmaged!

Thank you for reporting this issue. Fixing it would be a good improvement to the check. The solution suggested by @vbvictor seems reasonable.

Due to a lack of permissions I cannot assign myself to this issue. Perhaps @Szelethus can help with that?

**Comment by vbvictor** [comment]
You can still take this issue right-away and work on it.
